core/tests/protractor_desktop/ExplorationTranslationTab.js

```diff
@@ -131,11 +131,71 @@ describe('Exploration translation', function() {
     explorationEditorPage.saveChanges();
 
     explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.exitTutorial();
     explorationEditorTranslationTab.expectNumericalStatusToMatch(
       '(0/8)');
+
+    // click upload button
+    explorationEditorTranslationTab.openUploadAudioModal();
+    var audioToUpload = '../data/cafe.mp3';
+    var audioAbsolutePath = path.resolve(__dirname, audioToUpload);
+    var audioElem = element(by.css('[ng-class="inputFieldClassName"]'));
+    audioElem.sendKeys(audioAbsolutePath);
+    explorationEditorTranslationTab.saveUploadedAudio();
+    explorationEditorTranslationTab.expectNumericalStatusToMatch(
+      '(1/8)');
     users.logout();
   });
 
+  it('should provide correct status color ' +
+    'for each card(state) in Exploration Overview', function() {
+    users.createUser('user@correctstatus.com', 'correctStatus');
+    users.login('user@correctstatus.com');
+    workflow.createExploration();
+
+    explorationEditorMainTab.setStateName('First');
+    explorationEditorMainTab.setContent(forms.toRichText(
+      'This is first card.'));
+    explorationEditorMainTab.setInteraction('NumericInput');
+    explorationEditorMainTab.addResponse(
+      'NumericInput', forms.toRichText('This is feedback.'),
+      'Second', true, 'Equals', 6);
+    explorationEditorMainTab.moveToState('Second');
+    explorationEditorMainTab.setInteraction('Continue');
+    var responseEditor = explorationEditorMainTab.
+      getResponseEditor('default');
+    responseEditor.setDestination('Third', true, null);
+    explorationEditorMainTab.moveToState('Third');
+    explorationEditorMainTab.setInteraction('EndExploration');
+    explorationEditorMainTab.moveToState('First');
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.exitTutorial();
+    explorationEditorTranslationTab.openUploadAudioModal();
+    var audioToUpload = '../data/cafe.mp3';
+    var audioAbsolutePath = path.resolve(__dirname, audioToUpload);
+    var audioElem = element(by.css('[ng-class="inputFieldClassName"]'));
+    audioElem.sendKeys(audioAbsolutePath);
+    explorationEditorTranslationTab.saveUploadedAudio();
+    explorationEditorPage.navigateToMainTab();
+    explorationEditorMainTab.moveToState('Third');
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.openUploadAudioModal();
+    audioElem.sendKeys(audioAbsolutePath);
+    explorationEditorTranslationTab.saveUploadedAudio();
+    var colorOfFirstState = element.all(by.css(
+      'rect.protractor-test-node-background.init-node')).last().
```

Comment:
Move these selectors to `ExplorationEditorTranslationTab.js`. Also, you can look for graph related e2e selectors in `ExplorationEditorMainTab.js`

Reply:
done